tools/db_bench_tool.cc

```diff
@@ -7261,6 +7287,165 @@ class Benchmark {
     }
   }
 
+  void WaitForCompactionHelper(DBWithColumnFamilies& db) {
+    // This is an imperfect way of waiting for compaction. The loop and sleep
+    // is done because a thread that finishes a compaction job should get a
+    // chance to pickup a new compaction job.
+
+    std::vector<std::string> keys = { DB::Properties::kMemTableFlushPending,
+                                      DB::Properties::kNumRunningFlushes,
+                                      DB::Properties::kCompactionPending,
+                                      DB::Properties::kNumRunningCompactions };
+
+    fprintf(stdout, "waitforcompaction(%s): started\n", db.db->GetName().c_str());
+
+    while (true) {
+      bool retry = false;
+
+      for (const auto& k : keys) {
+        uint64_t v;
+        if (!db.db->GetIntProperty(k, &v)) {
+          fprintf(stderr, "waitforcompaction(%s): GetIntProperty(%s) failed\n",
+                 db.db->GetName().c_str(), k.c_str());
+          exit(1);
+        } else if (v > 0) {
+          fprintf(stdout, "waitforcompaction(%s): active(%s). Sleep 10 seconds\n",
+                  db.db->GetName().c_str(), k.c_str());
+          sleep(10);
+          retry = true;
+          break;
+        }
+      }
+
+      if (!retry) {
+        fprintf(stdout, "waitforcompaction(%s): finished\n",
+                db.db->GetName().c_str());
+        return;
+      }
+    }
+  }
+
+  void WaitForCompaction() {
+    // Give background threads a chance to wake
+    sleep(5);
+
+    // I am skeptical that this check race free. I hope that checking twice
```

Comment:
Checking for queued Env background jobs (```env->GetThreadPoolQueueLen()```) in ```WaitForCompactionhelper``` might make it more race free. I think a public API along the lines of ```DBImpl::TEST_WaitForCompact``` would be useful, but unfortunately we don't have one.